test/testmachinery/shoot/enable_rotate_certs_disable_test.go

```diff
@@ -0,0 +1,107 @@
+		// "should enable and disable the registry-cache extension" and "should enable extension, hibernate Shoot, reconcile Shoot, wake up Shoot, disable extension"
+		// tests and may present in the Node.
+		// So the current test will use ghcr.io/jitesoft/alpine:3.15.11 ghcr.io/jitesoft/alpine:3.16.9 and ghcr.io/jitesoft/alpine:3.17.9 images.
+		common.VerifyRegistryCache(parentCtx, f.Logger, f.ShootClient, common.GithubRegistryJitesoftAlpine31511Image, common.SleepInfinity)
+
+		By("Rotate certificates phase Preparing")
+		ctx, cancel = context.WithTimeout(parentCtx, 25*time.Minute)
+		defer cancel()
+		Expect(f.AnnotateShoot(ctx, f.Shoot, map[string]string{constants.GardenerOperation: constants.OperationRotateCAStart})).To(Succeed())
+		Expect(f.WaitForShootToBeReconciled(ctx, f.Shoot)).To(Succeed())
+		Expect(f.Shoot.Status.Credentials.Rotation.CertificateAuthorities.Phase).To(Equal(gardencorev1beta1.RotationPrepared))
+
+		By("Verify registry-cache works when phase is Prepared")
+		common.VerifyRegistryCache(parentCtx, f.Logger, f.ShootClient, common.GithubRegistryJitesoftAlpine3169Image, common.SleepInfinity)
+
+		By("Rotate certificates phase Completing")
+		ctx, cancel = context.WithTimeout(parentCtx, 10*time.Minute)
+		defer cancel()
+		Expect(f.AnnotateShoot(ctx, f.Shoot, map[string]string{constants.GardenerOperation: constants.OperationRotateCAComplete})).To(Succeed())
+		Expect(f.WaitForShootToBeReconciled(ctx, f.Shoot)).To(Succeed())
+		Expect(f.Shoot.Status.Credentials.Rotation.CertificateAuthorities.Phase).To(Equal(gardencorev1beta1.RotationCompleted))
+
+		By("Verify registry-cache works after CA rotation")
+		// After rotating the CA, the shoot client must be recreated, otherwise any attempt to use it will fail with::
+		// "tls: failed to verify certificate: x509: certificate signed by unknown authority"
+		shootClient, err := access.CreateShootClientFromAdminKubeconfig(ctx, f.GardenClient, f.Shoot)
+		Expect(err).NotTo(HaveOccurred())
+		common.VerifyRegistryCache(parentCtx, f.Logger, shootClient, common.GithubRegistryJitesoftAlpine3179Image, common.SleepInfinity)
+
+		By("Disable the registry-cache extension")
+		ctx, cancel = context.WithTimeout(parentCtx, 10*time.Minute)
+		defer cancel()
+		Expect(f.UpdateShoot(ctx, func(shoot *gardencorev1beta1.Shoot) error {
+			common.RemoveExtension(shoot, "registry-cache")
+
+			return nil
+		})).To(Succeed())
+
+		By("Verify registry configuration is removed")
+		ctx, cancel = context.WithTimeout(parentCtx, 2*time.Minute)
+		defer cancel()
+		common.VerifyHostsTOMLFilesDeletedForAllNodes(ctx, f.Logger, shootClient, []string{"ghcr.io"})
+	}, rotateCertTestTimeout, framework.WithCAfterTest(func(ctx context.Context) {
+		if f.Shoot.Status.Credentials != nil && f.Shoot.Status.Credentials.Rotation.CertificateAuthorities.Phase != gardencorev1beta1.RotationCompleted {
```

Comment:
Use the `v1beta1helper.GetShootCARotationPhase` func.

```suggestion
		if v1beta1helper.GetShootCARotationPhase(f.Shoot.Status.Credentials) != gardencorev1beta1.RotationCompleted  {
```